policies/rules/security/entry.go

```diff
@@ -0,0 +1,750 @@
+	FileBlockingProfiles     *util.MemberType `xml:"file-blocking,omitempty"`
+	WildfireAnalysisProfiles *util.MemberType `xml:"wildfire-analysis,omitempty"`
+	AntiVirusProfiles        *util.MemberType `xml:"virus,omitempty"`
+	AntiSpywareProfiles      *util.MemberType `xml:"spyware,omitempty"`
+	VulnerabilityProfiles    *util.MemberType `xml:"vulnerability,omitempty"`
+
+	Misc []generic.Xml `xml:",any"`
+}
+
+type qos struct {
+	Marking *qosMarking `xml:"marking,omitempty"`
+
+	Misc []generic.Xml `xml:",any"`
+}
+
+type qosMarking struct { // One of.
+	IpDscp                   *string `xml:"ip-dscp,omitempty"`
+	IpPrecedence             *string `xml:"ip-precedence,omitempty"`
+	FollowClientToServerFlow any     `xml:"follow-c2s-flow,omitempty"` // exist bool
+
+	Misc []generic.Xml `xml:",any"`
+}
+
+func SpecMatches(a, b *Entry) bool {
+	if a == nil && b == nil {
+		return true
+	} else if a == nil || b == nil {
+		return false
+	}
+
+	// omit compare: name
+	// omit compare: uuid
+
+	if !util.UnorderedListsMatch(a.SourceZones, b.SourceZones) {
+		return false
+	}
+
+	if !util.UnorderedListsMatch(a.DestinationZones, b.DestinationZones) {
+		return false
+	}
+
+	if !util.UnorderedListsMatch(a.SourceAddresses, b.SourceAddresses) {
+		return false
+	}
+
+	if !util.UnorderedListsMatch(a.SourceUsers, b.SourceUsers) {
+		return false
+	}
+
+	if !util.UnorderedListsMatch(a.DestinationAddresses, b.DestinationAddresses) {
+		return false
+	}
+
+	if !util.UnorderedListsMatch(a.Services, b.Services) {
+		return false
+	}
+
+	if !util.UnorderedListsMatch(a.Categories, b.Categories) {
+		return false
+	}
+
+	if !util.UnorderedListsMatch(a.Applications, b.Applications) {
+		return false
+	}
+
+	if !util.UnorderedListsMatch(a.SourceDevices, b.SourceDevices) {
+		return false
+	}
+
+	if !util.UnorderedListsMatch(a.DestinationDevices, b.DestinationDevices) {
+		return false
+	}
+
+	if !util.StringsMatch(a.Schedule, b.Schedule) {
+		return false
+	}
+
+	if !util.OrderedListsMatch(a.Tags, b.Tags) {
+		return false
+	}
+
+	if !util.BoolsMatch(a.NegateSource, b.NegateSource) {
+		return false
+	}
+
+	if !util.BoolsMatch(a.NegateDestination, b.NegateDestination) {
+		return false
+	}
+
+	if !util.BoolsMatch(a.Disabled, b.Disabled) {
+		return false
+	}
+
+	if !util.StringsMatch(a.Description, b.Description) {
+		return false
+	}
+
+	if !util.StringsMatch(a.GroupTag, b.GroupTag) {
+		return false
+	}
+
+	if a.Action != b.Action {
+		return false
+	}
+
+	if !util.BoolsMatch(a.IcmpUnreachable, b.IcmpUnreachable) {
+		return false
+	}
+
+	if !util.StringsMatch(a.Type, b.Type) {
+		return false
+	}
+
+	if !util.BoolsMatch(a.DisableServerResponseInspection, b.DisableServerResponseInspection) {
+		return false
+	}
+
+	if !util.BoolsMatch(a.LogStart, b.LogStart) {
+		return false
+	}
+
+	if !util.BoolsMatch(a.LogEnd, b.LogEnd) {
+		return false
+	}
+
+	// profile settings
+	if a.ProfileSettings == nil && b.ProfileSettings == nil {
```

Comment:
You create empty block, which doesn't make sense since there is nothing even to return here.

Reply:
We are looking for equality, so there are three situations:

* both are nil: they match
* one is nil while the other isn't: they don't match
* both are non-nil: you have to dive into the struct to verify equality

Thus this empty block is just the first situation, so there is nothing to do other than move on to other fields.

I understand it's weird to see an empty block, but it is doing the right thing here.  Is there something that you would like to see here instead?